### ISSUE TITLE
Redo with promises

### ISSUE THREAD

**Issue by shiffman** [CodingTrain/Mastodon-Bot#3]
Should I redo the earlier bot examples with promises? It's tricky b/c I'd like that to be available, but I used callbacks in the videos so would make it hard to follow along. Maybe I should make both available!

**Comment by WwaLlasS** [comment]
Hello, I will solve this issue, but I am not sure if I understood well, in my case the solution would be to make a separate file for example, but making the implementation of mastodon api based on promises, is that correct?

**Comment by hollyos** [comment]
If you or someone else is going to take the time to re-factor away from callbacks I would suggest skipping Promises and generator functions and jumping straight to **async/await**.

- https://hackernoon.com/6-reasons-why-javascripts-async-await-blows-promises-away-tutorial-c7ec10518dd9
- https://javascript.info/async-await
- https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/async_function
